src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -46,6 +46,10 @@ class DefaultStorageKeyAllocator
         return Format("acl/%x", static_cast<unsigned int>(index));
     }
 
+    // Group Message Counters
+    const char * GroupDataCounter() { return Format("gdc"); }
+    const char * GroupControlCounter() { return Format("gcc"); }
```

Comment:
- Suggest naming `SenderGlobalGroup[Data|Control]Counter()`
- Return `"gdc"` not `Format("gdc")`

Reply:
I had that thought, but that would not set the internal buffer, so consumers who then try to use it via `KeyName` would not work correctly, right?

Reply:
I went and triple checked:

ISO/IEC 14882:2017(E) (C++17)

5.13.5 String literals [lex.string]

> 16 Evaluating a string-literal results in a *string literal object with static storage duration*, initialized from
> the given characters as specified above. Whether all string literals are distinct (that is, are stored in
> nonoverlapping objects) and whether successive evaluations of a string-literal yield the same or a different
> object is unspecified. [ Note: The effect of attempting to modify a string literal is undefined. — end note ]

Therefore it's always OK to return a literal string where a `const char *` is returned, as the storage is present immutably within the program's static storage.

Reply:
That part is totally fine.  The part that would not work right is this:
```
    DefaultStorageKeyAllocater alloc;
    alloc.Format("gcc");
    // Use alloc.KeyName() here
```
which does work for other key types.  Maybe just documenting that would be enough, though.

Reply:
Yeah, but the Boris point's still stand. Consumer wouldn't be able to retrieve the key with the `KeyName` method since the internal buffer of the object wouldn't be set.